packages/wrangler/src/__tests__/generate.test.ts

```diff
@@ -62,30 +53,183 @@ describe("generate", () => {
+				"README.md": expect.stringContaining("Template: worker-typescript"),
+				"jest.config.json": expect.any(String),
+				"package.json": expect.stringContaining("@cloudflare/workers-types"),
+				src: expect.objectContaining({ "index.ts": expect.any(String) }),
+				test: expect.objectContaining({
+					"index.test.ts": expect.any(String),
+				}),
+				"tsconfig.json": expect.any(String),
+				"wrangler.toml": expect.any(String),
+			});
+			expect(fs.existsSync("my-worker-2")).toBe(false);
+
+			await expect(
+				runWrangler("generate my-worker worker-typescript")
+			).resolves.toBeUndefined();
+
+			expect(std.out).toStrictEqual(expect.stringContaining("my-worker-2"));
+
+			expect(readDirectory("my-worker-2")).toMatchObject<Directory>({
+				".git": expect.any(Object),
+				".gitignore": expect.any(String),
+				"README.md": expect.stringContaining("Template: worker-typescript"),
+				"jest.config.json": expect.any(String),
+				"package.json": expect.stringContaining("@cloudflare/workers-types"),
+				src: expect.objectContaining({ "index.ts": expect.any(String) }),
+				test: expect.objectContaining({
+					"index.test.ts": expect.any(String),
+				}),
+				"tsconfig.json": expect.any(String),
+				"wrangler.toml": expect.any(String),
+			});
+		});
 	});
 
-	it("auto-increments the worker directory name", async () => {
-		fs.mkdirSync("my-worker");
-		await expect(
-			runWrangler("generate my-worker some-template")
-		).resolves.toBeUndefined();
-
-		expect(createCloudflareMock).lastCalledWith(
-			"my-worker-1",
-			"some-template",
-			{ debug: false, force: false, init: true }
-		);
-
-		fs.mkdirSync("my-worker-1");
-
-		await expect(
-			runWrangler("generate my-worker some-template")
-		).resolves.toBeUndefined();
-
-		expect(createCloudflareMock).lastCalledWith(
-			"my-worker-2",
-			"some-template",
-			{ debug: false, force: false, init: true }
-		);
+	describe("cloning", () => {
+		it("clones a cloudflare template with sparse checkouts", async () => {
+			await expect(
+				runWrangler("generate my-worker worker-typescript")
+			).resolves.toBeUndefined();
+
+			expect(readDirectory("my-worker")).toMatchObject<Directory>({
+				".git": expect.any(Object),
+				".gitignore": expect.any(String),
+				"README.md": expect.stringContaining("Template: worker-typescript"),
+				"jest.config.json": expect.any(String),
+				"package.json": expect.stringContaining("@cloudflare/workers-types"),
+				src: expect.objectContaining({ "index.ts": expect.any(String) }),
+				test: expect.objectContaining({
+					"index.test.ts": expect.any(String),
+				}),
+				"tsconfig.json": expect.any(String),
+				"wrangler.toml": expect.any(String),
+			});
+		});
+
+		// mocking out calls to either `isGitInstalled` or `execa("git", ["--version"])`
+		// was harder than i thought, leaving this for now.
+		it.todo("clones a cloudflare template with full checkouts");
+
+		it("clones a user/repo template", async () => {
```

Comment:
Does this result in a git clone as well?
We should definitely circle back and look into whether we can mock aspects of this as well to avoid dependency upon network requests in our unit tests.